pkg/rulefmt/rulefmt.go

```diff
@@ -58,10 +60,50 @@ type RuleGroups struct {
 	Groups []RuleGroup `yaml:"groups"`
 }
 
+// RuleGroup is a list of sequentially evaluated recording and alerting rules.
+type RuleGroup struct {
+	Name     string         `yaml:"name"`
+	Interval model.Duration `yaml:"interval,omitempty"`
+	Rules    []RuleNode     `yaml:"rules"`
+}
+
+// RuleNode adds yaml.v3 layer to support line and column outputs for invalid rules.
+type RuleNode struct {
+	Record      yaml.Node         `yaml:"record,omitempty"`
+	Alert       yaml.Node         `yaml:"alert,omitempty"`
+	Expr        yaml.Node         `yaml:"expr"`
+	For         model.Duration    `yaml:"for,omitempty"`
+	Labels      map[string]string `yaml:"labels,omitempty"`
+	Annotations map[string]string `yaml:"annotations,omitempty"`
+}
+
+// ruleGroups type of yaml.v3 features in the groups field.
 type ruleGroups struct {
 	Groups []yaml.Node `yaml:"groups"`
 }
 
+// validateRules type for validating the groups inside the
```

Comment:
What's the rationale behind the public and private types here? I see that our public types are using `yaml.Node` which currently seems to have rendering problem per https://github.com/go-yaml/yaml/issues/575